src/codecs/codec.js

```diff
@@ -11,8 +11,12 @@
  * @param {(data:T) => Uint8Array} options.encode
  * @param {(bytes:Uint8Array) => T} options.decode
```

Comment:
Can you please add this to make sure that return complies with interface def

```suggestion
 * @returns {import('./interface'). BlockCodec<Code, T>}
```

Reply:
@Gozala note that this change forces further changes to `BlockCodec` because `encoder` and `decoder` are part of the return value now, so we then have to adjust the types to:

```diff
-export interface BlockCodec<Code extends number, T> extends BlockEncoder<Code, T>, BlockDecoder<Code, T> { }
+export interface BlockCodec<Code extends number, T> extends BlockEncoder<Code, T>, BlockDecoder<Code, T> {
+  encoder: BlockEncoder<Code, T>,
+  decoder: BlockDecoder<Code, T>
+}
```

Which I've done in my `fixup!`, but you may have opinions about that!

Reply:
Oh good catch, I forgot `Codec` was just `Encoder` + `Decoder` without those `encoder` / `decoder` props. Thanks!